### ISSUE TITLE
warn if TERM doesn't match terminal name

### ISSUE THREAD

**Issue by dankamongmen** [dankamongmen/notcurses#1335]
Now that we're doing all kinds of terminal-specific stuff, we might as well check the `TERM` value against the name of the terminal, if possible. This latter would presumably involve checking /proc or something similar -- not sure how it would be done on FreeBSD, etc. Like other helpful warnings, this would be inhibited by `NCOPTIONS_SUPRESS_BANNER`.

**Comment by dankamongmen** [comment]
obviously we're not talking exact match, but prefix match or more generally just some mapping.

**Comment by WSLUser** [comment]
Keep in mind that some terminals claim themselves as "xterm-blah-blah" even if they aren't one for one with xterm. Not sure if you should attempt to match to the terminal name with that being the case. Also a user can always change their TERM variable from the default. Usually that happens when experiencing some type of issue though that might get corrected by switching it to something else.

**Comment by dankamongmen** [comment]
yep yep -- i'm envisioning a list of terminal process names, and a list of known best `TERM` values, and warning when they don't match. if we don't know the best `TERM`, we wouldn't say anything. for terminals like `kmscon` where they explicitly recommend some `xterm` variant, we'd be looking for that variant. i intend no blind matching.

**Comment by dankamongmen** [comment]
See #1581, and perhaps proceed only via that route.

**Comment by dankamongmen** [comment]
So now that we're doing detection via query, a question comes up: how do we use that together with terminfo-provided information?

Some salient points:

* if we know the correct terminal via query, we can specify the best terminfo database *currently known*
* there might be a superior terminfo database definition later, that we don't know about
* the terminfo database might be obsolete relative to the actual terminal
* the terminfo database might be more correct than knowledge we encode about the terminal where they overlap

here are some "user stories":

* The user is using a modern terminal like Alacritty, but has foolishly exported `TERM=vt100`. Currently, we would mate discovered Sixel with a VT100 definition that doesn't even support color.
* The user is using XTerm compiled without RGB color support, has correctly exported `TERM=xterm-256color`, and has correctly not exported `COLORTERM`. Right now, we will properly not use RGB. If we unilaterally set `TERM=xterm-direct`, we would read the `rgb` terminfo capability, and unleash unsupported RGB.
* A terminal might not currently define `u7`, and thus we do not claim the ability to acquire the cursor position. But the terminal actually *does* support this capability. We hardcode it, and thus make it available to the user. The terminal then adds `u7`, implementing it in a different way than before. We continue to unleash the old escape, failing (if we got it from terminfo, we would send the proper escape).

**Comment by dankamongmen** [comment]
note that if we dropped terminfo, that would represent a fundamental cutdown in our dep stack.
